packages/xo-web/src/xo-app/backup-ng/_getSettingsWithNonDefaultValue.js

```diff
@@ -0,0 +1,25 @@
+const getSettingsWithNonDefaultValue = ({
+  compression = '',
+  mode,
+  settings: { '': globalSetting = {} } = {},
+}) => {
+  const settings = {}
+  if (globalSetting.reportWhen !== undefined) {
+    settings.reportWhen = globalSetting.reportWhen
+  }
+  if (globalSetting.concurrency > 0) {
+    settings.concurrency = globalSetting.concurrency
+  }
+  if (globalSetting.timeout > 0) {
+    settings.timeout = globalSetting.timeout
+  }
+  if (globalSetting.offlineSnapshot) {
+    settings.offlineSnapshot = globalSetting.offlineSnapshot
+  }
+  if (mode === 'full' && compression !== '') {
+    settings.compression = compression
+  }
+  return settings
```

Comment:
Why not using `omitBy`?

Reply:
The properties tests are different

Reply:
I prefer this code because more declarative:

```js
const DEFAULTS = {
  __proto__: null,

  compression: '',
  concurrency: 0,
  offlineSnapshot: false,
  timeout: 0,
}

const MODES = {
  __proto__: null,

  compression: 'full',
}

const getSettingsWithNonDefaultValue = ({
  compression = '',
  mode,
  settings: { '': globalSetting = {} } = {},
}) =>
  pickBy(globalSetting, (value, key) => {
    const settingMode = MODES[key]

    return (
      (settingMode === undefined || settingMode === mode) &&
      value !== undefined &&
      value !== DEFAULTS[key]
    )
  })
```